File: models/shift.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Shift represents a scheduled work period for a user at a specific location
type Shift struct {
	gorm.Model
	UserID     uint      `gorm:"not null;index"`
	LocationID uint      `gorm:"not null;index"`
	StartTime  time.Time `gorm:"not null;index"`
	EndTime    time.Time `gorm:"not null"`
	Notes      string    `gorm:"type:text"`
	Status     string    `gorm:"type:varchar(20);default:'scheduled'"` // e.g., scheduled, completed, cancelled
}

File: mobile/lib/models/shift.dart
class Shift {
  final int id;
  final String startTime;
  final String endTime;
  final String? clockInTime;
  final String? clockOutTime;
  final String status;

  Shift({
    required this.id,
    required this.startTime,
    required this.endTime,
    this.clockInTime,
    this.clockOutTime,
    required this.status,
  });

  factory Shift.fromJson(Map<String, dynamic> json) {
    return Shift(
      id: json['ID'],
      startTime: json['StartTime'],
      endTime: json['EndTime'],
      clockInTime: json['ClockInTime'],
      clockOutTime: json['ClockOutTime'],
      status: json['Status'] ?? 'scheduled',
    );
  }
}
